Makefile

```diff
@@ -46,9 +46,22 @@ else
 	include fs/Makefile
 	include disk/Makefile
 	#include virtio/Makefile
-	include sample/helloworld/Makefile
-	include sample/test/Makefile
-	include sample/test/fails/Makefile
+	include sample/Makefile
 	include setjmp/Makefile
 
 endif
+
+PROGS_INSTALL_FS := $(addprefix $(PREFIX_FS)bin/phoenix-rtos-tests/, $(TEST_HARNESS_PROGS) $(TEST_UNITY_PROGS))
+
+$(addprefix $(PREFIX_PROG), $(TEST_HARNESS_PROGS)): $(PREFIX_PROG)% : $(PREFIX_O)%.o
+	$(LINK)
+
+$(addprefix $(PREFIX_PROG), $(TEST_UNITY_PROGS)): $(PREFIX_PROG)% : $(PREFIX_O)%.o $(PREFIX_A)unity.a
+	$(LINK)
+
+$(PROGS_INSTALL_FS): $(PREFIX_FS)bin/phoenix-rtos-tests/% : $(PREFIX_PROG)%
```

Comment:
- we should probably install stripped versions of the binaries in the filesystem
- I don't like the idea of having subdirs in `/bin`. I don't have any subdirs on my host computer there :no_good: 
- I think we either should install all tests in `/bin` or - If You really want to - we can make it `/test`
- we don't really need to be afraid of overwriting tests (they should have unique names), I think `make` might warn us if there are multiple rules for the same target (I don't know if it will work with the current rules) - You can easily test it locally :)

Reply:
Now tests are placed into `/bin` without subdirs. Also I changed a little bit building strategy. Let me know what you think.